README.md

```diff
@@ -20,3 +20,27 @@ randomly without memory instead of systematically trying them?
 
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. Add your answer to this markdown file.
+
+///\
+In the best case, the input data is already sorted. The permutationsort function only needs to check once and directly find the first permutation, which is the sorted permutation. It needs to iterate n times. In this case, the time complexity is O(n)\
+
+In the worst case, for an array of length n, there are n choices for the first element, n-1 choices for the second element, n-2 choices for the third element, and so on, until n-n.\
+The sorted permutation is the last one among all permutations, which requires traversing all permutations, and the time complexity is O(n*n!)\
+
+When randomly generating permutations, as the size of the array approaches infinity, the probability of finding the sorted one in n! permutations is 1/n!, the probability of randomly generating a sorted array approaches 0, and the running time approaches infinity.\
+
+
+###
+Sources: Chatgpt provided me with some structure of how to generate all permutations in the generatePermutations function and receive the array and the starting position of the permutation.Just some basic syntax and basic code logic.\
+Chatgpt give me a prompt,including a pseudo code: "Benchmark condition: start == arr.length - 1, add a copy of arr to permutations, traverse i from start to arr.length - 1, swap arr[start] and arr[i], generatePermutations(arr, start + 1) recursive call, backtrack (swap back to the original position)"
```

Comment:
I don't understand what you mean. It sounds like ChatGPT prompted you and not the other way round.